### ISSUE TITLE
IllegalArgumentException: cannot change field "Communication:From" from index options=NONE to inconsistent index options=DOCS_AND_FREQS_AND_POSITIONS

### ISSUE THREAD

**Issue by lfcnassif** [sepinf-inc/IPED#1007]
Sometimes processing an ufdr with master aborts with error below from lucene-9.0.0:
```
Caused by: java.lang.IllegalArgumentException: cannot change field "ufed:Data" from index options=DOCS_AND_FREQS_AND_POSITIONS to inconsistent index options=NONE
	at org.apache.lucene.index.FieldInfo.verifySameIndexOptions(FieldInfo.java:245)
	at org.apache.lucene.index.FieldInfos$FieldNumbers.verifySameSchema(FieldInfos.java:420)
	at org.apache.lucene.index.FieldInfos$FieldNumbers.addOrGet(FieldInfos.java:357)
	at org.apache.lucene.index.FieldInfos$Builder.add(FieldInfos.java:667)
	at org.apache.lucene.index.FieldInfos$Builder.add(FieldInfos.java:625)
	at org.apache.lucene.index.IndexingChain.initializeFieldInfo(IndexingChain.java:669)
	at org.apache.lucene.index.IndexingChain.processDocument(IndexingChain.java:610)
	at org.apache.lucene.index.DocumentsWriterPerThread.updateDocuments(DocumentsWriterPerThread.java:239)
	at org.apache.lucene.index.DocumentsWriter.updateDocuments(DocumentsWriter.java:432)
	at org.apache.lucene.index.IndexWriter.updateDocuments(IndexWriter.java:1530)
	at org.apache.lucene.index.IndexWriter.updateDocument(IndexWriter.java:1815)
	at org.apache.lucene.index.IndexWriter.addDocument(IndexWriter.java:1468)
	at dpf.sp.gpinf.indexer.process.task.IndexTask.process(IndexTask.java:173)
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processMonitorTimeout(AbstractTask.java:253)
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:184)
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:229)
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:192)
	at dpf.sp.gpinf.indexer.process.task.AbstractTask.sendToNextTask(AbstractTask.java:229)
```

I'll try to fix "ufed:Data" field data type to see if it helps.

**Comment by lfcnassif** [comment]
Happened again:
```
Caused by: java.lang.IllegalArgumentException: cannot change field "ufed:Value" from index options=DOCS_AND_FREQS_AND_POSITIONS to inconsistent index options=NONE
at org.apache.lucene.index.FieldInfo.verifySameIndexOptions(FieldInfo.java:245) ~[lucene-core-9.0.0.jar:9.0.0 0b18b3b965cedaf5eb129aa41243a44c83ca826d - jpountz - 2021-12-01 14:23:49]
at org.apache.lucene.index.FieldInfos$FieldNumbers.verifySameSchema(FieldInfos.java:420) ~[lucene-core-9.0.0.jar:9.0.0 0b18b3b965cedaf5eb129aa41243a44c83ca826d - jpountz - 2021-12-01 14:23:49]
at org.apache.lucene.index.FieldInfos$FieldNumbers.addOrGet(FieldInfos.java:357) ~[lucene-core-9.0.0.jar:9.0.0 0b18b3b965cedaf5eb129aa41243a44c83ca826d - jpountz - 2021-12-01 14:23:49]
at org.apache.lucene.index.FieldInfos$Builder.add(FieldInfos.java:667) ~[lucene-core-9.0.0.jar:9.0.0 0b18b3b965cedaf5eb129aa41243a44c83ca826d - jpountz - 2021-12-01 14:23:49]
at org.apache.lucene.index.FieldInfos$Builder.add(FieldInfos.java:625) ~[lucene-core-9.0.0.jar:9.0.0 0b18b3b965cedaf5eb129aa41243a44c83ca826d - jpountz - 2021-12-01 14:23:49]
at org.apache.lucene.index.IndexingChain.initializeFieldInfo(IndexingChain.java:669) ~[lucene-core-9.0.0.jar:9.0.0 0b18b3b965cedaf5eb129aa41243a44c83ca826d - jpountz - 2021-12-01 14:23:49]
at org.apache.lucene.index.IndexingChain.processDocument(IndexingChain.java:610) ~[lucene-core-9.0.0.jar:9.0.0 0b18b3b965cedaf5eb129aa41243a44c83ca826d - jpountz - 2021-12-01 14:23:49]
at org.apache.lucene.index.DocumentsWriterPerThread.updateDocuments(DocumentsWriterPerThread.java:239) ~[lucene-core-9.0.0.jar:9.0.0 0b18b3b965cedaf5eb129aa41243a44c83ca826d - jpountz - 2021-12-01 14:23:49]
at org.apache.lucene.index.DocumentsWriter.updateDocuments(DocumentsWriter.java:432) ~[lucene-core-9.0.0.jar:9.0.0 0b18b3b965cedaf5eb129aa41243a44c83ca826d - jpountz - 2021-12-01 14:23:49]
at org.apache.lucene.index.IndexWriter.updateDocuments(IndexWriter.java:1530) ~[lucene-core-9.0.0.jar:9.0.0 0b18b3b965cedaf5eb129aa41243a44c83ca826d - jpountz - 2021-12-01 14:23:49]
at org.apache.lucene.index.IndexWriter.updateDocument(IndexWriter.java:1815) ~[lucene-core-9.0.0.jar:9.0.0 0b18b3b965cedaf5eb129aa41243a44c83ca826d - jpountz - 2021-12-01 14:23:49]
at org.apache.lucene.index.IndexWriter.addDocument(IndexWriter.java:1468) ~[lucene-core-9.0.0.jar:9.0.0 0b18b3b965cedaf5eb129aa41243a44c83ca826d - jpountz - 2021-12-01 14:23:49]
```
Reopening util this is properly fixed.

**Comment by lfcnassif** [comment]
I was finally able to reproduce this easily with artificial data:
`IllegalArgumentException: cannot change field "html:propriedade" from index options=DOCS_AND_FREQS_AND_POSITIONS to inconsistent index options=NONE`

I think I know what is going on and now I can test possible solutions.

**Comment by lfcnassif** [comment]
Of course, the issue is a race condition when we try to auto detect the type of an unknown field. If that field appears in 2 or more items being processed at the same time, the auto detect logic can run multiple times for that field simultaneously. If one of the field values is a number for one item and a string for another, we are trying to index that field using different types.

The Lucene message is a bit misleading because it prints `IndexType.NONE`, but, surprisingly to me, that is true for numeric fields (IntPoint, FloatPoint, DoublePoint)...

**Comment by hauck-jvsh** [comment]
Just to know, was this happening in 3.1x?

**Comment by lfcnassif** [comment]
I think so, but old Lucene didn't use to detect and abort with that situation.

**Comment by brunofs-br** [comment]
I got an error trying to open a multicase, is it related to this issue ?

.\hdd_note\iped\lib\iped-search-app.jar -multicases .


`2022-08-12 11:53:43	[INFO]	[app.ui.AppMain]			Indexador e Processador de Evidências Digitais 4.0.1`
...
```
2022-08-12 12:07:22	[INFO]	[engine.data.IPEDMultiSource]			Opening MultiReader...
java.lang.IllegalArgumentException: cannot change field "audio:xmpDM:discNumber" from index options=NONE to inconsistent index options=DOCS_AND_FREQS_AND_POSITIONS
	at org.apache.lucene.index.FieldInfo.verifySameIndexOptions(FieldInfo.java:254)
	at org.apache.lucene.index.FieldInfo.verifySameSchema(FieldInfo.java:215)
	at org.apache.lucene.index.FieldInfos$Builder.add(FieldInfos.java:692)
	at org.apache.lucene.index.FieldInfos$Builder.add(FieldInfos.java:664)
	at org.apache.lucene.index.FieldInfos.getMergedFieldInfos(FieldInfos.java:210)
	at iped.engine.lucene.SlowCompositeReaderWrapper.<init>(SlowCompositeReaderWrapper.java:122)
	at iped.engine.lucene.SlowCompositeReaderWrapper.wrap(SlowCompositeReaderWrapper.java:97)
	at iped.engine.data.IPEDMultiSource.openIndex(IPEDMultiSource.java:178)
	at iped.engine.data.IPEDMultiSource.init(IPEDMultiSource.java:129)
	at iped.engine.data.IPEDMultiSource.<init>(IPEDMultiSource.java:79)
```

**Comment by lfcnassif** [comment]
> I got an error trying to open a multicase, is it related to this issue ?

I think so and also related to #661. Your error is possibly caused by a new property being detected as different types in different cases, because it could have different values in different cases. It is not easy to fix, I'll have to think how solve this problem...

**Comment by lfcnassif** [comment]
I'll keep this closed and try to fix the multicase issue on #661.

**Comment by lfcnassif** [comment]
Hi @brunofs-br, I implemented a quick and dirty workaround to ignore this problem when opening multicases. Please test the snapshot below with your multicase:
https://github.com/sepinf-inc/IPED/actions/runs/2863511709

You just need to replace the lib/iped-engine-4.0.x.jar file into the main case of your multicase (the one used to open it)
